### PR TITLE
Make CC port configurable in KafkaRebalanceAssemblyOperator

### PR DESCRIPTION
**Pull request by fvaleri**
Related to this [PR comment](https://github.com/strimzi/strimzi-kafka-operator/pull/10056#discussion_r1591395028).

### CODE REVIEW

**Comment by ppatierno** [comment]
@fvaleri why there was the need to make the CC port configurable? Just a genuine question ... maybe tests were failing with the default CC port not free and the need to find a free one? I haven't never seen a failure for that.

**Comment by fvaleri** [comment]
> maybe tests were failing with the default CC port not free and the need to find a free one? I haven't never seen a failure for that.

In general, CC "standard" port (9090) is fairly common, and could be already in use by another process on your machine. This change avoids this potential port conflict, and makes these tests more robust. It's already implemented that way in other tests.

**Review by ppatierno — APPROVED** [review]
LGTM. Thanks!

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java [hunk 1]
```diff
@@ -156,21 +156,37 @@ public class KafkaRebalanceAssemblyOperator
...
-
     private final ConfigMapOperator configMapOperator;
+    private int cruiseControlPort;
+
+    /**
+     * @param vertx The Vertx instance
+     * @param supplier Supplies the operators for different resources
+     * @param config Cluster Operator configuration
+     */
+    public KafkaRebalanceAssemblyOperator(Vertx vertx,
```
[the remark this answers is not in the document]
Comment (fvaleri) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java:167]: This is used in Main, but I think you mean the new constructor on line 179, where we can use the default modifier. Done.

[the remark this answers is not in the document]
Comment (scholzj) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java:167]: Yeah, sorry. I meant the new constructor.

[the remark this answers is not in the document]
Comment (fvaleri) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java:167]: Np, should be good now.

**Review by scholzj — APPROVED** [review]

**Pull request merged**